plugins-builtin/src/main/scala/za/co/absa/enceladus/plugins/builtin/errorsender/mq/KafkaErrorSenderPluginImpl.scala

```diff
@@ -0,0 +1,191 @@
+import za.co.absa.enceladus.utils.error.ErrorMessage.ErrorCodes
+
+import scala.util.{Failure, Success, Try}
+
+
+case class KafkaErrorSenderPluginImpl(connectionParams: KafkaConnectionParams,
+                                      keySchemaRegistryConfig: Map[String, String],
+                                      valueSchemaRegistryConfig: Map[String, String]) extends PostProcessor {
+
+  private val log = LogManager.getLogger(classOf[KafkaErrorSenderPluginImpl])
+
+  override def close(): Unit = {}
+
+  /**
+   * When data is ready, the error record(s) are pusblished to kafka.
+   *
+   * @param dataFrame error data only.
+   * @param params    Additional key/value parameters provided by Enceladus.
+   */
+  override def onDataReady(dataFrame: DataFrame, params: Map[String, String]): Unit = {
+    if (!SchemaUtils.fieldExists(ColumnNames.enceladusRecordId, dataFrame.schema)) {
+      throw new IllegalStateException(
+        s"${this.getClass.getName} requires ${ColumnNames.enceladusRecordId} column to be present in the dataframe!"
+      )
+    }
+
+    val errorSenderParams = Try(ErrorSenderPluginParams.fromMap(params)) match {
+      case Success(params) => params
+      case Failure(e) =>throw new IllegalArgumentException(
+        s"Incompatible parameter map supplied for ${KafkaErrorSenderPluginImpl.getClass.getName}: $params", e
+      )
+    }
+
+    val dfWithErrors = getIndividualErrors(dataFrame, errorSenderParams)
+
+    if (dfWithErrors.isEmpty) {
+      log.info("No errors to be sent to kafka.")
+    } else {
+      val count = dfWithErrors.count()
+      log.info(s"Sending $count errors to kafka topic ${connectionParams.topicName} ...")
+
+      Try {
+        val forKafkaDf = prepareDataForKafka(dfWithErrors, errorSenderParams)
+        sendErrorsToKafka(forKafkaDf)
+      } match {
+        case Success(_) => log.info(s"$count errors successfully sent to kafka topic ${connectionParams.topicName}")
+        case Failure(e) =>
+          log.error("Sending errors to kafka unsuccessful due to: ", e)
+      }
+    }
+  }
+
+  /**
+   * Processes the `dataFrame` - errors are exploded (one error = one line) and filtered to conform to the error source (standardization/conformance)
+   * @param dataFrame standardization/conformance output
+   * @param params plugin processing parameters
+   * @return DF with exploded errors and corresponding to the given error source
+   */
+  def getIndividualErrors(dataFrame: DataFrame, params: ErrorSenderPluginParams): DataFrame = {
+    implicit val singleErrorStardardizedEncoder = Encoders.product[SingleErrorStardardized]
+    implicit val dceErrorEncoder = Encoders.product[DceError]
+
+    val allowedErrorCodes = KafkaErrorSenderPluginImpl.errorCodesForSource(params.sourceId)
+
+    val stdErrors = dataFrame
+      .filter(size(col("errCol")) > 0)
+      // only keep columns that are needed for the actual error publishing
```

Comment:
```suggestion
      // only keep rows that are needed for the actual error publishing
```

Reply:
no, yeah, well... The comment was meant for the subsequent code, not for the prior one.

The fact that you are suggesting a change like this, however, shows that it is misleading, so will touch it up a bit :D

Reply:
Done.